src/main/java/org/zalando/nakadi/service/subscription/SubscriptionService.java

```diff
@@ -0,0 +1,71 @@
+import org.zalando.nakadi.exceptions.NakadiException;
+import org.zalando.nakadi.repository.EventTypeRepository;
+import org.zalando.nakadi.repository.TopicRepository;
+import org.zalando.nakadi.service.subscription.model.Partition;
+import org.zalando.nakadi.service.subscription.zk.ZkSubscriptionClient;
+import org.zalando.nakadi.service.subscription.zk.ZkSubscriptionClientFactory;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Component
+public class SubscriptionService {
+
+    private final EventTypeRepository eventTypeRepository;
+    private final ZkSubscriptionClientFactory zkSubscriptionClientFactory;
+    private final TopicRepository topicRepository;
+
+    @Autowired
+    public SubscriptionService(final ZkSubscriptionClientFactory zkSubscriptionClientFactory,
+                               final TopicRepository topicRepository,
+                               final EventTypeRepository eventTypeRepository) {
+        this.zkSubscriptionClientFactory = zkSubscriptionClientFactory;
+        this.topicRepository = topicRepository;
+        this.eventTypeRepository = eventTypeRepository;
+    }
+
+    public List<SubscriptionEventTypeStats> createSubscriptionStat(final Subscription subscription) {
+        final ZkSubscriptionClient zkSubscriptionClient =
+                zkSubscriptionClientFactory.createZkSubscriptionClient(subscription.getId());
+        final Partition[] partitions = zkSubscriptionClient.listPartitions();
+        return subscription.getEventTypes().stream()
+                .map(ExceptionWrapper.wrapFunction(eventTypeRepository::findByNameO))
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .map(eventType -> {
+                    final Set<SubscriptionEventTypeStats.Partition> statPartitions = Arrays.stream(partitions)
+                            .filter(partition -> eventType.getTopic().equals(partition.getKey().topic))
+                            .map(ExceptionWrapper.wrapFunction(
+                                    partition -> createPartition(zkSubscriptionClient, partition)))
+                            .collect(Collectors.toSet());
+                    return new SubscriptionEventTypeStats(eventType.getName(), statPartitions);
+                })
+                .collect(Collectors.toList());
+    }
+
+    private SubscriptionEventTypeStats.Partition createPartition(final ZkSubscriptionClient zkSubscriptionClient,
+                                                                 final Partition partition) throws NakadiException {
+        final String partitionName = partition.getKey().partition;
+        final String partitionState = partition.getState().description;
+        final String partitionSession = partition.getSession();
+        final TopicPartition topicPartition = topicRepository.getPartition(partition.getKey().topic,
```

Comment:
This is a very time-consuming operation, calling it separately for each partition can be very time-consuming.
I suggest to create a new method in TopicRepository to fetch this data for multiple topic/partitions in one call. In general KafkaTopicRepository.listPartitions(...) is almost what you need, the difference in your case is that your topic/partition pairs can be of different topics.